name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/editor/ClientInput.java

```diff
@@ -1,511 +1,605 @@
+        else
+            dialog.setFilterPath(clientFile != null ? clientFile.getAbsoluteFile().getParent()
+                            : System.getProperty("user.home")); //$NON-NLS-1$
+
+        String path = dialog.open();
+        if (path == null)
+            return;
+
+        // again make sure the extension is correct as the user might have
+        // changed it in the save dialog
+        if (extension != null && !path.endsWith('.' + extension))
+            path += '.' + extension;
+
+        File localFile = new File(path);
+        char[] password = null;
+
+        if (ClientFactory.isEncrypted(localFile))
+        {
+            PasswordDialog pwdDialog = new PasswordDialog(shell);
+            if (pwdDialog.open() != Window.OK)
+                return;
+            password = pwdDialog.getPassword().toCharArray();
+        }
+
+        clientFile = localFile;
+        label = localFile.getName();
+        char[] pwd = password;
+
+        BusyIndicator.showWhile(shell.getDisplay(), () -> {
+            try
+            {
+                ClientFactory.save(client, clientFile, encryptionMethod, pwd);
+                storePreferences(true);
+
+                broker.post(UIConstants.Event.File.SAVED, clientFile.getAbsolutePath());
+                setDirty(false, false);
+                listeners.forEach(ClientInputListener::onSaved);
+            }
+            catch (IOException e)
+            {
+                PortfolioPlugin.log(e);
+                ErrorDialog.openError(shell, Messages.LabelError, e.getMessage(),
+                                new Status(IStatus.ERROR, PortfolioPlugin.PLUGIN_ID, e.getMessage(), e));
+            }
+        });
+    }
+
+    /**
+     * autoSave is called from AutoSaveJob and uses the file generated during startup,
+     * if set in preferences
+     */
+    public void autoSave()
+    {
+        if (clientFile != null)
+        {
+            // generate the same name used when creating the initial autosave file
+            // see @scheduleAutoSaveJob 
+            String suffix = "autosave"; //$NON-NLS-1$
+            String filename = clientFile.getPath();
+            int l = filename.lastIndexOf('.');
+            String backupName = l > 0 ? filename.substring(0, l) + '.' + suffix + filename.substring(l)
+                            : filename + '.' + suffix;
+
+            try
+            {
+                File tempFile = new File(backupName);
+                
+                // file should exist, because a copy is created, during setup of the job
+                if (!tempFile.exists())
+                {
+                    createBackup(clientFile, suffix);
+                }
+                ClientFactory.save(client, tempFile, null, null);
+            }
+            catch (IOException e)
+            {
+                PortfolioPlugin.log(e);
+            }
+        }
+    }
+    
+    public long getAutoSavePrefs()

```

Comment:
Die Methode muss nicht ```public``` sein, oder?

Reply:
Das ist richtig. In der Heartbeat Version habe ich es von AutoSaveJob aus aufgerufen, deshalb ist das stehen geblieben. Wird geändert.